ulib/FStar.List.Tot.Base.fst

```diff
@@ -273,28 +273,27 @@ predicate [f] *)
 (** [filter f l] returns [l] with all elements [x] such that [f x]
 does not hold removed. Requires, at type-checking time, [f] to be a
 pure total function.  Named as in: OCaml, Coq *)
-val filter : #a: Type -> f:(a -> Tot bool) -> l: list a -> Tot (m:list a{forall x. memP x m ==> f x})
+val filter : #a: Type -> f:(a -> Tot bool) -> l: list a -> Tot (list a)
 let rec filter #a f = function
   | [] -> []
   | hd::tl -> if f hd then hd::filter f tl else filter f tl
 
 (** Postcondition on [filter f l]: for any element [x] of [filter f l],
-[f x] holds. Requires, at type-checking time, [f] to be a pure total
-function.*)
+[f x] holds, and x was a member of l. Requires, at type-checking time,
+[f] to be a pure total function. *)
+let rec mem_filter_forall (#a: Type) (f: (a -> Tot bool)) (l: list a)
+    : Lemma (forall x. memP x (filter f l) <==> (f x /\ memP x l))
+            [SMTPat (filter f l)] =
+  match l with
```

Comment:
Hi @chandradeepdey , same comment as in the other PR. I believe you could prove this too by just manipulating classical connectives with `mem_filter` (it may require implies intro as well).